normalize-git-url.js

```diff
@@ -18,7 +18,16 @@ module.exports = function normalize (u) {
 
   var returnedUrl
   if (parsed.pathname.match(/\/?:/)) {
-    returnedUrl = u.replace(/^(?:git\+)?ssh:\/\//, '').replace(/#[^#]*$/, '')
+    if (u.match(/^git\+https?/) && parsed.pathname.match(/\/?:[^0-9]/)) {
+      returnedUrl = u.replace(/^(.*:[^:]+):(.*)/, '$1/$2')
+      console.warn('deprecated: ' + u + ' is using an' +
```

Comment:
npm proper has actual support for deprecation ([`lib/util/depr-check.js`](https://github.com/npm/npm/blob/2.x/lib/utils/depr-check.js)), and is probably where that warning should be coming from. I agree that using `console.warn` here doesn't feel particularly clean. I think it would be worth taking the time to get the warning into the CLI.